src/ngAria/aria.js

```diff
@@ -377,6 +377,25 @@ ngAriaModule.directive('ngShow', ['$aria', function($aria) {
             });
           }
         }
+
+        if (elem[0].nodeName === 'A') {
+          if ($aria.config('bindRoleForClick') && !attr.href && !attr.xlinkHref && !attr.role) {
+              elem.attr('role', 'link');
+          }
+          if ($aria.config('bindKeypress')) {
+            elem.on('keypress', function(event) {
+              var keyCode = event.which || event.keyCode;
+              var hasHref = (attr.href || attr.xlinkHref);
+              if ((keyCode === 32 || (keyCode === 13 && !hasHref)) && !attr.ngKeypress) {
+                elem[0].click();
```

Comment:
Taking https://github.com/angular/angular.js/pull/13996#discussion_r53084886 into consideration, we should only concern ourselves with `<a>` without `href`/`xlinkHref`, which would remove the ambiguity.

And then, the only difference would be that `fn(...)` would not trigger other `click` handlers on the element, while `.click()` would. And since (1) the user has already specified `ngClick` on the element and (2) it's good to be consistent with other non-button elements (e.g. `<div>` or `<li>`), I think it's better to use `fn(...)`.